lib/internal/modules/typescript.js

```diff
@@ -136,7 +137,10 @@ function stripTypeScriptModuleTypes(source, filename) {
 function addSourceMap(code, sourceMap) {
   // TODO(@marco-ippolito) When Buffer.transcode supports utf8 to
   // base64 transformation, we should change this line.
-  const base64SourceMap = internalBinding('buffer').btoa(sourceMap);
+  // The base64 encoding should be https://datatracker.ietf.org/doc/html/rfc4648#section-4,
+  // not base64url https://datatracker.ietf.org/doc/html/rfc4648#section-5. See data url
+  // spec https://tools.ietf.org/html/rfc2397#section-2.
+  const base64SourceMap = Buffer.from(sourceMap).toString('base64');
```

Comment:
Isn't transcode a better solution?

Reply:
The intent here is `string -> string`. `transcode` is buffer -> buffer, I don't see the advantage here with `transcode` like operation `string -> buffer trancode -> buffer -> string`.

Reply:
Ah, I see. Thanks!